### PR TITLE
feat: add verifier capability

### PR DESCRIPTION
**Pull request by Artemkaaas**
# Summary of Changes

The chanegs in the PR add Verifier capability into Aries Mobile Agent according to the design document provided here: https://github.com/hyperledger/aries-mobile-agent-react-native/pull/649

~~The PR is still under the active development (we are testing the workflow, doing code refacoring, writing tests for new components)  but we decided to open it now to start the review process ( gahter feedback). (the branch is based on the `main`).~~

Overview of key changes:
* Now user can enable verifier capability in the developer settings. Using this capability Verifier can request data from another phone via two flos:
  * Verifier generate QR containing request which can be scanned and replied by any other party (conenctionless case).
  * Verifier sends request through the already established connection to a specific user.    
* A verifier can see the history of used proof request tempaltes.
* Connection chat evaluated to show the history of events between parties (credentials and proofs) in addition to sent messages. 
* Design of Contact List was changed to be more generic (previously it was tied to the chat functionality)  
* We added an ability to create connection invitations (by default. the feature is hidden under developer options) in order to test proof request flow through the established connection.
* Important: The interface of OCA bundle resolver was change in order to add support for request templates and shared proof. Now it's credential agnostic. It's breaking change but I see that it was broken in some recent [PR](https://github.com/hyperledger/aries-mobile-agent-react-native/pull/634) as well.

# Pull Request Checklist

Tick all boxes below to demonstrate that you have completed the respective task. If the item does not apply to your this PR **check it anyway** to make it apparent that there's nothing to do.

- [x] All commits contain a DCO `Signed-off-by` line (we use the [DCO GitHub app](https://github.com/apps/dco) to enforce this);
- [ ] Updated LICENSE-3RD-PARTY.md for any added dependencies or vendored components;
- [x] Updated documentation as needed for changed code and new or modified features;
- [x] Added sufficient [tests](../__tests__/) so that overall code coverage is not reduced.

If you have _any_ questions to _any_ of the points above, just **submit and ask**! This checklist is here to _help_ you, not to deter you from contributing!

Pro Tip 🤓

- Read our [contribution guide](../CONTRIBUTING.md) at least once; it will save you a few review cycles!
- Your PR will likely not be reviewed until all the above boxes are checked and all automated tests have passed.

_PR template adapted from the Python attrs project._

### CODE REVIEW

**Comment by Artemkaaas** [comment]
@hyperledger/aries-mobile-agent-react-native-committers 
Is there any feedback on this PR? Has someone started looking into it?

File: core/App/screens/AttemptLockout.tsx
Comment (bryce-mcmath) [inline core/App/screens/AttemptLockout.tsx]: What's the reasoning behind enabling users to navigate back to the lockout screen?

File: core/App/components/chat/ChatMessage.tsx
Comment (bryce-mcmath) [inline core/App/components/chat/ChatMessage.tsx]: I don't think we follow the pattern of starting interfaces with `I` in the rest of the app, so we might not want to start doing that here

File: core/App/screens/PINCreate.tsx
Comment (bryce-mcmath) [inline core/App/screens/PINCreate.tsx]: This is out of sync with the main branch I think, most recently this was refactored into it's own component

File: core/App/screens/PINCreate.tsx
Comment (bryce-mcmath) [inline core/App/screens/PINCreate.tsx]: Yeah I think there is some syncing issues here. These are reverts of some of our most recent changes

File: core/App/screens/CommonDecline.tsx
Comment (bryce-mcmath) [inline core/App/screens/CommonDecline.tsx]: I'm not sure if this is true but I seem to recall us not wanting to send problem reports for declined creds/proofs for some reason. I'll double check and get back to you.

**Review by bryce-mcmath — CHANGES_REQUESTED** [review]
The biggest issue I can see is that there is some sync issues with the main branch, where you are adding reverts of recent changes, not sure what caused that. It might be hard to track them all down. Left some other minor comments

File: core/App/screens/AttemptLockout.tsx
Comment (Artemkaaas) [reply core/App/screens/AttemptLockout.tsx]: Thanks for pointing.  There was hard rebace  of our branch on the upstream due to DCO issue in some our commits. Now I see that some places where merged incorrectly.  
I went through the changes once again and corrected all places.

File: core/App/screens/CommonDecline.tsx
Comment (Artemkaaas) [reply core/App/screens/CommonDecline.tsx]: It just looks strange now when Verifier sends Reqeust which always stays in `Requested` state even if Prover rejected it. Moreover, according to the RFC Prover should send Problem Report when rejects a request.

File: core/App/components/chat/ChatMessage.tsx
Comment (Artemkaaas) [reply core/App/components/chat/ChatMessage.tsx]: Renamed to `ExtandedChatMessage`

File: core/App/screens/PINCreate.tsx
Comment (Artemkaaas) [reply core/App/screens/PINCreate.tsx]: Fixed

File: core/App/screens/PINCreate.tsx
Comment (Artemkaaas) [reply core/App/screens/PINCreate.tsx]: Fixed

File: core/App/screens/ListContacts.tsx
Comment (jcdrouin21) [inline core/App/screens/ListContacts.tsx]: The background of the list should not be the same as the background of the items, right? Currently, both use `ColorPallet.brand.secondaryBackground`

Comment (Artemkaaas) [reply core/App/screens/ListContacts.tsx]: The list of credentials looks different because there are cards.  
We made a list of contacts to look like the list of credential attributes now and list of request attributes/predicates before switching to cards

File: core/App/screens/ProofRequestDetails.tsx
Comment (jcdrouin21) [inline core/App/screens/ProofRequestDetails.tsx]: Should the primary button be above the secondary button ?

File: core/App/screens/ProofRequesting.tsx
Comment (jcdrouin21) [inline core/App/screens/ProofRequesting.tsx]: Same here. Should the primary button be above the secondary button ?

File: core/App/screens/Chat.tsx
Comment (jcdrouin21) [inline core/App/screens/Chat.tsx]: In french, the `ProofRequestSent` message is too long and goes off the screen. 
image

File: core/App/screens/ProofRequestDetails.tsx
Comment (Artemkaaas) [reply core/App/screens/ProofRequestDetails.tsx]: Swapped

File: core/App/screens/ProofRequesting.tsx
Comment (Artemkaaas) [reply core/App/screens/ProofRequesting.tsx]: Swapped

File: core/App/screens/Chat.tsx
Comment (Artemkaaas) [reply core/App/screens/Chat.tsx]: Fixed

File: packages/legacy/core/App/components/chat/ChatEvent.tsx
Comment (jleach) [inline packages/legacy/core/App/components/chat/ChatEvent.tsx]: Will the `userLabel` be hard coded in one of the translation files or does it come from the connection? I don't think we need to translated here because we won't know what it is.

File: packages/legacy/core/App/components/chat/ChatEvent.tsx
Comment (jleach) [inline packages/legacy/core/App/components/chat/ChatEvent.tsx]: Same.

File: packages/legacy/core/App/components/chat/ChatMessage.tsx
Comment (jleach) [inline packages/legacy/core/App/components/chat/ChatMessage.tsx]: I think you mean `them`.

File: packages/legacy/core/App/components/chat/ChatMessage.tsx
Comment (jleach) [inline packages/legacy/core/App/components/chat/ChatMessage.tsx]: Other places in the app use `{formatTime(message.createdAt)}` where `formatTime` is is one of our util functions. Are you able to use this for consistency? This way may or may not be better but I think consistency is more important for now.

File: packages/legacy/core/App/components/chat/ChatMessage.tsx
Comment (jleach) [inline packages/legacy/core/App/components/chat/ChatMessage.tsx:88]: I think this will need a `testID` and `accessibilityLabel`.

File: packages/legacy/core/App/components/listItems/ContactListItem.tsx
Comment (jleach) [inline packages/legacy/core/App/components/listItems/ContactListItem.tsx]: testID and accessibilityLabel

File: packages/legacy/core/App/components/misc/QRRenderer.tsx
Comment (jleach) [inline packages/legacy/core/App/components/misc/QRRenderer.tsx]: I think in TS you want the nil coalescing operator `??`

```
const qrSize = size ?? windowDimensions.width - 80
```

File: packages/legacy/core/App/components/record/RecordField.tsx
Comment (jleach) [inline packages/legacy/core/App/components/record/RecordField.tsx:42]: Because these all return I don't think you need the else if else. Removing it will make the code more readable.

```
if x {
  return
}

if y {
  return
}

return z
```

File: packages/legacy/core/App/components/record/RecordField.tsx
Comment (jleach) [inline packages/legacy/core/App/components/record/RecordField.tsx:107]: The prop type for field is `Attribute` so the case may not be needed here.

File: packages/legacy/core/App/components/views/CollapsibleView.tsx
Comment (jleach) [inline packages/legacy/core/App/components/views/CollapsibleView.tsx]: Let's move this into a separate file and import like other SVGs for consistency unless there is a good reason for an in-line SVG.

File: packages/legacy/core/App/hooks/credentials.ts
Comment (jleach) [inline packages/legacy/core/App/hooks/credentials.ts:5]: This `fn` already exists in the [react native hooks for AFJ](https://cs.github.com/?scopeName=All+repos&scope=&q=repo%3Ahyperledger%2Faries-framework-javascript-ext+useCredentialsByConnectionId). Wondering if we need it in Bifold?

File: packages/legacy/core/App/hooks/proofs.ts
Comment (jleach) [inline packages/legacy/core/App/hooks/proofs.ts:5]: This `fn` already exists in the [react native hooks for AFJ](https://cs.github.com/hyperledger/aries-framework-javascript-ext/blob/3712cb2d88ba9a2ea6ada9cb0e57351c3987f872/packages/react-hooks/src/ProofProvider.tsx?q=repo%3Ahyperledger%2Faries-framework-javascript-ext+useProofsByConnectionId#L28). Wondering if we need it in Bifold?

File: packages/legacy/core/App/screens/ProofRequesting.tsx
Comment (jleach) [inline packages/legacy/core/App/screens/ProofRequesting.tsx]: Remove dead code?

File: packages/legacy/core/App/screens/ProofRequesting.tsx
Comment (jleach) [inline packages/legacy/core/App/screens/ProofRequesting.tsx:32]: I don't think this component is used. Its not exported and commented out below. Let's remove it.

File: packages/legacy/core/App/screens/ListProofRequests.tsx
Comment (jleach) [inline packages/legacy/core/App/screens/ListProofRequests.tsx:78]: Needs an accessibilityLabel.

File: packages/legacy/core/App/screens/ProofDetails.tsx
Comment (jleach) [inline packages/legacy/core/App/screens/ProofDetails.tsx:225]: Is this return needed?

File: packages/legacy/core/App/screens/ProofRequestUsageHistory.tsx
Comment (jleach) [inline packages/legacy/core/App/screens/ProofRequestUsageHistory.tsx]: Will need testId and accessabilityId

**Review by jleach — CHANGES_REQUESTED** [review]

File: packages/legacy/core/App/components/chat/ChatEvent.tsx
Comment (Artemkaaas) [reply packages/legacy/core/App/components/chat/ChatEvent.tsx]: Fixed

File: packages/legacy/core/App/components/chat/ChatEvent.tsx
Comment (Artemkaaas) [reply packages/legacy/core/App/components/chat/ChatEvent.tsx]: Fixed

File: packages/legacy/core/App/components/chat/ChatMessage.tsx
Comment (Artemkaaas) [reply packages/legacy/core/App/components/chat/ChatMessage.tsx]: Fixed

File: packages/legacy/core/App/components/chat/ChatMessage.tsx
Comment (Artemkaaas) [reply packages/legacy/core/App/components/chat/ChatMessage.tsx]: Fixed

File: packages/legacy/core/App/components/chat/ChatMessage.tsx
Comment (Artemkaaas) [reply packages/legacy/core/App/components/chat/ChatMessage.tsx:88]: Fixed

File: packages/legacy/core/App/components/listItems/ContactListItem.tsx
Comment (Artemkaaas) [reply packages/legacy/core/App/components/listItems/ContactListItem.tsx]: Fixed

File: packages/legacy/core/App/screens/ProofRequestUsageHistory.tsx
Comment (Artemkaaas) [reply packages/legacy/core/App/screens/ProofRequestUsageHistory.tsx]: Fixed

File: packages/legacy/core/App/screens/ProofDetails.tsx
Comment (Artemkaaas) [reply packages/legacy/core/App/screens/ProofDetails.tsx:225]: Linter warns that return type must be either boolean or undefined or null.  If we return this statement the return type will be void

File: packages/legacy/core/App/components/misc/QRRenderer.tsx
Comment (Artemkaaas) [reply packages/legacy/core/App/components/misc/QRRenderer.tsx]: Fixed

File: packages/legacy/core/App/components/record/RecordField.tsx
Comment (Artemkaaas) [reply packages/legacy/core/App/components/record/RecordField.tsx:42]: Fixed

File: packages/legacy/core/App/screens/ProofRequesting.tsx
Comment (Artemkaaas) [reply packages/legacy/core/App/screens/ProofRequesting.tsx:32]: Removed `ProcessingView` component

File: packages/legacy/core/App/components/record/RecordField.tsx
Comment (Artemkaaas) [reply packages/legacy/core/App/components/record/RecordField.tsx:107]: Linter warn becasue the type is Field

File: packages/legacy/core/App/components/views/CollapsibleView.tsx
Comment (Artemkaaas) [reply packages/legacy/core/App/components/views/CollapsibleView.tsx]: Removed as unused

File: packages/legacy/core/App/screens/ProofRequesting.tsx
Comment (Artemkaaas) [reply packages/legacy/core/App/screens/ProofRequesting.tsx]: Removed

File: packages/legacy/core/App/hooks/credentials.ts
Comment (Artemkaaas) [reply packages/legacy/core/App/hooks/credentials.ts:5]: Export of these function were missing in the hooks library.  I opened the PR: https://github.com/hyperledger/aries-framework-javascript-ext/pull/205

File: packages/legacy/core/App/hooks/proofs.ts
Comment (Artemkaaas) [reply packages/legacy/core/App/hooks/proofs.ts:5]: Export of these function were missing in the hooks library.  I opened the PR: https://github.com/hyperledger/aries-framework-javascript-ext/pull/205

File: packages/legacy/core/App/screens/ListProofRequests.tsx
Comment (Artemkaaas) [reply packages/legacy/core/App/screens/ListProofRequests.tsx:78]: Fixed

**Review by jleach — DISMISSED** [review]
Thanks for the changes. LGTM. Just need to fix the merge conflicts and @bryce-mcmath's change.

**Comment by Artemkaaas** [comment]
> Thanks for the changes. LGTM. Just need to fix the merge conflicts and @bryce-mcmath's change.

Merged and solved conflicts

**Review by jleach — APPROVED** [review]

**Comment by jleach** [comment]
@jcdrouin21 @bryce-mcmath I believe you requested changes so this PR will need your approval to move fwd.

**Review by bryce-mcmath — APPROVED** [review]

**Pull request merged**
